### PR TITLE
Add support for spatial audio

### LINKED ISSUE #1: Component to supply camera position to spacial system in FMOD

### LINKED ISSUE #16: Move studio.get_event out of EventReader loop
https://github.com/Salzian/bevy_fmod/blob/e7f088aa870a70da53312ea7079f82013f4a1126/src/plugin.rs#L68

Fetching the event every time it's played seems inefficient. We should fetch the event references on startup (maybe even some fancy compile-time validation) and only pass references to the `play_incoming_events` system.

### LINKED ISSUE #21: Configure FMOD to use righthanded coordinate system
FMOD may also be configured to use a right-handed coordinate system by passing `FMOD_INIT_3D_RIGHTHANDED` to `System::init`. This will make it so that we don't have to do any of our own conversions.

### PR DESCRIPTION
**Pull request by GitGhillie**
See the added example spatial.rs.

Adds components to handle FMOD events in a Bevy way, by implementing AudioSinkPlayback and storing a reference to the FMOD event.

Closes #1 
Closes #21
Closes #16

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,12 +9,18 @@ license = "MIT OR Apache-2.0"
 
 [dependencies]
 anyhow = "1.0.75"
-bevy = { version = "0.11.2", default-features = false }
+bevy = { version = "0.11.2", default-features = false, features = ["bevy_audio"] }
```
Comment (Salzian) [inline Cargo.toml:12]: We should add a test crate that depends on this crate to run unit tests and ensure the crate functions correctly as a library. With just examples, I assume this mistake wouldn't have been obvious.

File: Cargo.toml [hunk 2]
```diff
@@ -9,12 +9,18 @@ license = "MIT OR Apache-2.0"
...
 bevy_mod_sysfail = "3.0.0"
 libfmod = "2.206.2"
+once_cell = "1.18.0"
```
Comment (Salzian) [inline Cargo.toml:15]: I can't see this being imported anywhere. Where is this used?

File: src/components/audio_listener.rs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#[derive(Component, Default)]
+pub struct AudioListener;
+
+impl AudioListener {
+    pub(crate) fn update_3d_attributes(
+        query: Query<(&Velocity, &GlobalTransform), With<AudioListener>>,
+        studio: Res<FmodStudio>,
+    ) {
+        match query.get_single() {
+            Ok((velocity, transform)) => {
```
Comment (Salzian) [inline src/components/audio_listener.rs:16]: Good catch.

File: Cargo.toml (hunk 1 above)
Comment (GitGhillie) [reply Cargo.toml:12]: I haven't done much unit testing before (let alone with Rust) but that sounds reasonable

**Review by Salzian — APPROVED** [review]

**Pull request merged**